tests/periph_uart/main.c

```diff
@@ -154,6 +216,45 @@ static int cmd_init(int argc, char **argv)
     return 0;
 }
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+static int cmd_mode(int argc, char **argv)
+{
+    int dev;
+    uart_data_bits_t data_bits;
+    uart_parity_t  parity;
+    uart_stop_bits_t  stop_bits;
+
+    if (argc < 5) {
+        printf("usage: %s <dev> <data bits> <parity> <stop bits>\n", argv[0]);
+        return 1;
+    }
+
+    dev = parse_dev(argv[1]);
+    if (dev < 0) {
+        return 1;
+    }
+    data_bits = parse_data_bits(argv[2]);
```

Comment:
this does potentially break: there is no guarantee what the actual value returned by this function might look like (it might be defined freely on a per-CPU basis). So the check in the line below might not work reliable for any platform...

I suggest simply to remove the static function and put the data bit parsing code here directly!

Reply:
I have moved all parsing actions to `cmd_mode`.